src/main/webapp/app/course/manage/course-management.service.ts

```diff
@@ -155,21 +157,27 @@ export class CourseManagementService {
             .pipe(map((res: EntityArrayResponseType) => this.convertDateArrayFromServer(res)));
     }
 
+    /**
+     * returns the course with the provided unique identifier for the tutor dashboard
+     * @param courseId - the id of the course
+     */
+    getForTutors(courseId: number): Observable<EntityResponseType> {
+        const url = `${this.resourceUrl}/${courseId}/for-tutor-dashboard`;
+        return this.http
+            .get<Course>(url, { observe: 'response' })
+            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
+    }
+
     /**
      * returns the course with the provided unique identifier for the tutor dashboard
      * @param courseId - the id of the course
      * @param examId - Id of the exam when it is in exam mode, otherwise undefined
      */
-    getForTutors(courseId: number, examId: number): Observable<EntityResponseType> {
-        if (examId) {
-            return this.http
-                .get<Course>(`${this.resourceUrl}/${courseId}/exam/${examId}/for-exam-tutor-dashboard`, { observe: 'response' })
-                .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
-        } else {
-            return this.http
-                .get<Course>(`${this.resourceUrl}/${courseId}/for-tutor-dashboard`, { observe: 'response' })
-                .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
-        }
+    getExamWithExercises(courseId: number, examId: number): Observable<ExamEntityResponseType> {
+        const url = `${this.resourceUrl}/${courseId}/exams/${examId}/for-exam-tutor-dashboard`;
+        return this.http.get<Exam>(url, { observe: 'response' });
+        // TODO: convert date from server
+        // .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
```

Comment:
Why is this a TODO?
```suggestion
        return this.http.get<Exam>(url, { observe: 'response' })
            .pipe(map((res: EntityResponseType) => this.convertDateFromServer(res)));
```

Reply:
Types were not matching because this was implemented in the course service. Moved this to the exam service.